core/storage/file/gae_models_test.py

```diff
@@ -0,0 +1,229 @@
+        file_model.commit(feconf.SYSTEM_COMMITTER_ID, [])
+
+        file_model.content = 'file_contents'
+        commit_cmds = [{'cmd': 'edit'}]
+        file_model.commit(feconf.SYSTEM_COMMITTER_ID, commit_cmds)
+        retrieved_model = file_models.FileModel.get_version(
+            'exp_id1', 'path/to/file1.png', 2)
+        self.assertEqual(file_model.key, retrieved_model.key)
+        self.assertEqual(retrieved_model.content, 'file_contents')
+
+    def test_file_model_snapshot_includes_file_model_content(self):
+        file_model = file_models.FileModel.create(
+            'exp_id1', 'path/to/file1.png')
+
+        self.assertIsNone(file_model.content)
+
+        file_model.content = 'file_contents'
+        file_model.commit(feconf.SYSTEM_COMMITTER_ID, [])
+
+        self.assertEqual(file_model.content, 'file_contents')
+
+    def test_get_new_id_raises_not_implemented_error(self):
+        with self.assertRaises(NotImplementedError):
+            file_models.FileModel.get_new_id('entity1')
+
+    def test_create_creates_model_with_correct_id_and_deleted_status_as_false(
+            self):
+        file_model1 = file_models.FileModel.create(
+            'exp_id1', 'path/to/file1.png')
+        self.assertEqual(file_model1.deleted, False)
+        self.assertEqual(file_model1.id, '/exp_id1/path/to/file1.png')
+
+        file_model2 = file_models.FileModel.create(
+            'exp_id1', '/exp_id1/path/to/file2.png')
+        self.assertEqual(file_model2.deleted, False)
+        self.assertEqual(file_model2.id, '/exp_id1/path/to/file2.png')
+
+    def test_get_model_with_model_present_returns_the_correct_model(self):
+        file_model = file_models.FileModel.create(
+            'exp_id1', 'path/to/file1.png')
+        file_model.commit(feconf.SYSTEM_COMMITTER_ID, [])
+        retrieved_model = file_models.FileModel.get_model(
+            'exp_id1', 'path/to/file1.png')
+
+        self.assertEqual(retrieved_model, file_model)
+
+    def test_get_model_non_strict_with_no_model_present_returns_none(self):
+        retrieved_model = file_models.FileModel.get_model(
+            'exp_id1', 'path/to/file2.png')
+        self.assertIsNone(retrieved_model)
+
+    def test_get_model_strict_with_no_model_present_raises_erro(self):
+        with self.assertRaisesRegexp(
+            base_models.BaseModel.EntityNotFoundError, (
+                'Entity for class FileModel '
+                'with id /exp_id1/path/to/file2.png not found')):
+            file_models.FileModel.get_model(
+                'exp_id1', 'path/to/file2.png', True)
+
+    def test_commit_updates_version_of_stored_model(self):
+        file_model = file_models.FileModel.create(
+            'exp_id1', 'path/to/file1.png')
+        self.assertEqual(file_model.version, 0)
+
+        file_model.commit(feconf.SYSTEM_COMMITTER_ID, [])
+        self.assertEqual(file_model.version, 1)
+
+    def test_get_version_with_version_present_returns_correct_model(self):
```

Comment:
Maybe add another commit in this method to verify that it's actually selecting the first version (versus the second, which has different content). It's not much different of a test, but it might help more clearly indicate that this is really working version selection (so in cases where version selection always picks the last version, this test would then fail).